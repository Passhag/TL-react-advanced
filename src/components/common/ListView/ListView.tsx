import * as React from 'react';

interface Props {
ViewItem: React.ComponentClass<any> | React.SFC<any>;
  items: (Object & { id: string | number })[];
}

const ListView: React.SFC<Props> = ({ ViewItem, items }): JSX.Element => {
  const listItems = items.map(item => (
    <li className="list-group-item" key={item.id}>
      <ViewItem item={item} />
    </li>
  ));

  return (
    <ul className="list-group">
      {listItems}
    </ul>
  );
};

export default ListView;
