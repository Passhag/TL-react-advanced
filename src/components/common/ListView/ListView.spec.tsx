import * as React from 'react';
import { shallow, configure } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import ListView from './ListView';

configure({ adapter: new ReactSixteenAdapter() });

describe('Components::ListView', () => {
  it('should render correctly', () => {
    const ViewItem = () => <li />;
    const wrapper = shallow(
      <ListView
        items={[{ id: 1 }, { id: 2 }]}
        ViewItem={ViewItem}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render viewItems correctly', () => {
    const ViewItem = item => (
      <li>
        {item.id}
      </li>
    );
    const wrapper = shallow(
      <ListView
        items={[{ id: 1 }, { id: 2 }]}
        ViewItem={ViewItem}
      />
    );

    expect(wrapper.find('li')).toHaveLength(2);
  });
});
