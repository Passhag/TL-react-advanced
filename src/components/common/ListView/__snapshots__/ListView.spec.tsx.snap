// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components::ListView should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListView
    ViewItem={[Function]}
    items={
        Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="list-group-item"
>
          <ViewItem
                    item={
                              Object {
                                        "id": 1,
                                      }
                    }
          />
</li>,
        <li
          className="list-group-item"
>
          <ViewItem
                    item={
                              Object {
                                        "id": 2,
                                      }
                    }
          />
</li>,
      ],
      "className": "list-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <ViewItem
            item={
                        Object {
                                    "id": 1,
                                  }
            }
/>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": <ViewItem
            item={
                        Object {
                                    "id": 2,
                                  }
            }
/>,
          "className": "list-group-item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "item": Object {
              "id": 2,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="list-group-item"
>
            <ViewItem
                        item={
                                    Object {
                                                "id": 1,
                                              }
                        }
            />
</li>,
          <li
            className="list-group-item"
>
            <ViewItem
                        item={
                                    Object {
                                                "id": 2,
                                              }
                        }
            />
</li>,
        ],
        "className": "list-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <ViewItem
              item={
                            Object {
                                          "id": 1,
                                        }
              }
/>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "id": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <ViewItem
              item={
                            Object {
                                          "id": 2,
                                        }
              }
/>,
            "className": "list-group-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "item": Object {
                "id": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
